File: src/main/java/com/meepoffaith/hextrapats/casting/actions/OpSplitList.java
package com.meepoffaith.hextrapats.casting.actions;

import at.petrak.hexcasting.api.casting.eval.CastingEnvironment;
import at.petrak.hexcasting.api.casting.iota.Iota;
import at.petrak.hexcasting.api.casting.iota.ListIota;
import com.meepoffaith.hextrapats.casting.bases.ConstMediaActionBase;
import com.meepoffaith.hextrapats.casting.bases.HexIotaStack;

import java.util.ArrayList;
import java.util.List;

public class OpSplitList extends ConstMediaActionBase{
    public int argc = 2;
    public long mediaCost = 0L;

    @Override
    public List<? extends Iota> execute(HexIotaStack stack, CastingEnvironment ctx){
        ArrayList<Iota> list = stack.getJUSTAList(0);
        int index = stack.getPositiveIntUnder(1, list.size());
        return List.of(new ListIota(list.subList(0, index)), new ListIota(list.subList(index, list.size())));
    }
}

File: src/common/main/kotlin/com/meepoffaith/hextrapats/casting/actions/lists/OpSplitList.kt
package com.meepoffaith.hextrapats.casting.actions.lists

import at.petrak.hexcasting.api.casting.castables.ConstMediaAction
import at.petrak.hexcasting.api.casting.eval.CastingEnvironment
import at.petrak.hexcasting.api.casting.getList
import at.petrak.hexcasting.api.casting.getPositiveIntUnder
import at.petrak.hexcasting.api.casting.iota.Iota
import at.petrak.hexcasting.api.casting.iota.ListIota

object OpSplitList : ConstMediaAction {
    override val argc = 2
    override fun execute(args: List<Iota>, env: CastingEnvironment): List<Iota> {
        val list = args.getList(0, argc).toMutableList()
        val index = args.getPositiveIntUnder(1, list.size + 1, argc)
        return listOf(ListIota(list.subList(0, index)), ListIota(list.subList(index, list.size)))
    }
}
